src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1043,28 +1082,43 @@ MPS_Tensor MPS::state_vec_as_MPS(uint_t first_index, uint_t last_index) const
 	}
 	// now temp is a tensor of 2^n matrices of size 1X1
 	temp.mul_Gamma_by_right_Lambda(right_lambda);
-
 	return temp;
 }
 
-void MPS::full_state_vector(cvector_t& statevector) const
-{
-  MPS_Tensor mps_vec = state_vec_as_MPS(0, num_qubits_-1);
-  uint_t length = 1ULL << num_qubits_;   // length = pow(2, num_qubits_)
-  statevector.resize(length);
+void MPS::full_state_vector(cvector_t& statevector){
```

Comment:
This can be `const` , right?

Reply:
This can no longer by const because it involves moving the qubits before computation. This is true for all methods that do a computation directly on the MPS structure, without first copying it to a temporary structure.